redfish-core/lib/fabric_adapters.hpp

```diff
@@ -173,11 +173,36 @@ inline void
     });
 }
 
+inline void linkAsPCIeDevice(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                             const std::string& fabricAdapterPath)
+{
+    const std::string pcieDeviceName =
+        sdbusplus::message::object_path(fabricAdapterPath).filename();
+
+    if (pcieDeviceName.empty())
+    {
+        BMCWEB_LOG_ERROR ("Failed to find / in pcie device path");
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    nlohmann::json& deviceArray = asyncResp->res.jsonValue["Links"]["PCIeDevices"];
+    deviceArray = nlohmann::json::array();
+
+    deviceArray.push_back(
+        {{"@odata.id",
+          "/redfish/v1/Systems/system/PCIeDevices/" + pcieDeviceName}});
```

Comment:
See https://github.com/openbmc/bmcweb/blob/master/COMMON_ERRORS.md#13-complete-replacement-of-the-response-object.

Or, similar pattern is https://github.com/openbmc/bmcweb/blob/41b25aa3a541a0f92dee53b89064f655ac5e11cb/redfish-core/lib/systems.hpp#L3406

It is suggested to right like

```
nlohmann::json::object_t item;
item["@odata.id"] = boost::urls::format("/redfish/v1/Systems/system/PCIeDevices/{}", pcieDeviceName);
deviceArray.emplace_back( std::move(item) );
```

Reply:
Ack